app/screens/image_preview/video_preview.js

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    StyleSheet,
+    View
+} from 'react-native';
+import Video from 'react-native-video';
+import RNFetchBlob from 'react-native-fetch-blob';
+
+import EventEmitter from 'mattermost-redux/utils/event_emitter';
+
+import VideoControls, {PLAYER_STATE} from 'app/components/video_controls';
+import {DeviceTypes} from 'app/constants/';
+
+import Downloader from './downloader.ios';
+
+const {VIDEOS_PATH} = DeviceTypes;
+
+export default class VideoPreview extends PureComponent {
+    static propTypes = {
+        deviceHeight: PropTypes.number.isRequired,
+        deviceWidth: PropTypes.number.isRequired,
+        file: PropTypes.object,
+        onFullScreen: PropTypes.func,
+        onSeeking: PropTypes.func,
+        theme: PropTypes.object
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isLoading: true,
+            isFullScreen: false,
+            playerState: PLAYER_STATE.PAUSED,
+            paused: true,
+            currentTime: 0,
+            duration: 0,
+            path: null,
+            showDownloader: true
+        };
+    }
+
+    componentWillMount() {
+        EventEmitter.on('stop-video-playback', this.stopPlayback);
+        this.initializeComponent();
+    }
+
+    componentWillUnmount() {
+        EventEmitter.off('stop-video-playback', this.stopPlayback);
+    }
+
+    async initializeComponent() {
+        const {file} = this.props;
+        const path = `${VIDEOS_PATH}/${file.id}.${file.extension}`;
+        const exist = await RNFetchBlob.fs.exists(path);
+
+        if (exist) {
+            this.setState({path});
+        }
+    }
+
+    exitFullScreen = () => {
+        this.setState({isFullScreen: false});
+        this.props.onFullScreen();
+    };
+
+    enterFullScreen = () => {
+        this.setState({isFullScreen: true});
+        this.props.onFullScreen();
+    };
+
+    onDownloadSuccess = () => {
+        const {file} = this.props;
+        const path = `${VIDEOS_PATH}/${file.id}.${file.extension}`;
+
+        this.setState({showDownloader: false, path});
+    };
+
+    onEnd = () => {
+        if (this.state.isFullScreen) {
+            this.props.onFullScreen();
+        }
+
+        this.setState({playerState: PLAYER_STATE.ENDED, isFullScreen: false, paused: true});
+    };
+
+    onError = () => {
+        //TODO: Show an error
```

Comment:
Can an alert be added here to at least show something to the user if there is an error?